help_config.go

```diff
@@ -177,8 +177,22 @@ func newHelpConfigKeyFromField(f reflect.StructField) (helpConfigKey, error) {
 		}
 	case reflect.Bool:
 		h.typ = "bool"
+	case reflect.Map:
+		switch f.Type.Key().Kind() {
+		case reflect.String:
+			switch f.Type.Elem().Kind() {
+			case reflect.String:
+				h.typ = "table of strings"
+			case reflect.Int:
+				h.typ = "table of ints"
+			default:
+				return h, fmt.Errorf("config key %q: unsupported type table of %s", f.Name, f.Type.Elem())
+			}
+		default:
+			return h, fmt.Errorf("config key %q: unsupported map with key type %s", f.Name, f.Type.Key())
```

Comment:
I know the TOML standard calls these _tables_ but I think we should stick to `map` in the help message

Reply:
I was opting for the table naming because a table in the TOML world implicitly uses strings as keys. If we say `map of strings` and `map of ints` we may not be sufficiently precise about the type. Maybe we should use `map of strings to strings` and `map of strings to ints`. I am open to suggestions.

Reply:
`map of strings to strings` and `map of strings to ints` is good